spx-gui/src/apis/aigc.ts

```diff
@@ -2,11 +2,170 @@
  * @desc AI-related APIs of spx-backend
  */
 
-import { client } from './common'
+import type { AssetData, AssetType } from './asset'
+import { client, type FileCollection } from './common'
 
 export async function matting(imageUrl: string) {
   const result = (await client.post('/aigc/matting', { imageUrl }, { timeout: 20 * 1000 })) as {
     imageUrl: string
   }
   return result.imageUrl
 }
+
+export type AIAssetData<T extends AssetType = AssetType> = {
+  /** Globally unique ID */
+  id: string
+  // /** Asset Category */
+  // category: string
+  /** Asset Type */
+  assetType: T
+  /** Files the asset contains */
+  files?: FileCollection
+  /** Preview URL for the asset, e.g., a gif for a sprite */
+  preview?: string
+  /** Creation time */
+  cTime: string
+  status: AIGCStatus
+}
+
+export const AssetSourceKey = Symbol('AssetSource')
+export type AssetOrAIAsset =
+  | (AssetData & { [AssetSourceKey]?: 'public' })
+  | (AIAssetData & { [AssetSourceKey]?: 'ai' })
+
+export interface CreateAIImageParams {
+  keyword: string
+  category: string | string[]
+  assetType: AssetType
+  width?: number
+  height?: number
+}
+
+const mockAIImage = {
+  imageUri: [
+    'kodo://goplus-builder-static-test/files/FvEYel08GXL60vuviffq9sW-9xZs/IMG20230406220355.jpg',
+    'kodo://goplus-builder-static-test/files/FtzNex_e0lYKKn7S52HTG42d9Mm9-6906',
+    'kodo://goplus-builder-static-test/files/FvbBWmTdYLqyx_Mn_wCF6KAhfZXn-5443'
+  ]
+}
+
+/**
+ * Generate AI image with given keyword and category
+ *
+ * WARNING: This API has not been implemented yet. It will return a mock result.
+ */
+export async function generateAIImage({
+  keyword,
+  category,
+  assetType,
+  width,
+  height
+}: CreateAIImageParams) {
+  return new Promise<{ imageJobId: string }>((resolve) => {
+    setTimeout(() => {
+      resolve({ imageJobId: `mock-${keyword}-${Math.random().toString(36).slice(2)}` })
+    }, 1000)
+  })
+  // const result = (await client.post(
+  //   '/aigc/image',
+  //   { keyword, category, assetType, width, height },
+  //   { timeout: 20 * 1000 }
+  // )) as {
+  //   imageJobId: string
+  // }
+  // return result.imageJobId
+}
```

Comment:
If it's still needed, keep it. If it's not needed, delete it. Don't leave large blocks of commented-out code.

Please check for similar comments in the current PR

Reply:
Ok.

These are the actual functional code that should be kept at the end, and the code above them simply returns mock data first. 

Now I have uncommented them.